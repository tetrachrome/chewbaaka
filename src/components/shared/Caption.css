/**
 * Caption.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 09, 2020
 * Updated  : Jul 30, 2020
 */

div.CaptionContainer {
  margin: 0 auto;
  max-width: 100%;
}

div.CaptionContainer_DebugColor {
  background-color: cadetblue;
}

div.IconContainer {
  float: left;
  margin-right: 4px;
}

div.IconContainer_DebugColor {
  background-color: coral;
}

p.CaptionText {
  /** Specify a height to make sure variable content
   * does not cause layout size change (i.e., page jump).
   */
  max-height: 75px;
  text-align: left;
  font-family: Arial, Helvetica, sans-serif;
  margin-right: 5px;
}
