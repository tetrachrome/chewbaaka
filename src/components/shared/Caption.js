/**
 * Caption.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 09, 2020
 * Updated  : Jul 30, 2020
 */

/**
 * Caption
 *
 * `Caption` is a component that illustrates
 * a piece of caption text.
 *
 * Props
 *
 *  - `caption`: The caption text.
 *
 *  - `width` (optional): Optional width of the caption text block in pixels.
 */

import React from 'react'

import { Icon } from "semantic-ui-react"

import 'semantic-ui-css/semantic.min.css'

import {
  getElementStyleClassName,
  getElementStyleClassNames
} from '../../styling/styling'

import './ContentPageSharedStyles.css'

import './Caption.css'

export default function Caption(props) {
  const containerDimensionStyle = {
    maxWidth: props.width,
  };

  return (
    <div
      style={containerDimensionStyle}
      className={getElementStyleClassName("CaptionContainer")}
      data-testid="CaptionComponentTestId"
    >
      <div className={getElementStyleClassName("IconContainer")}>
        <Icon name="image" />
      </div>
      <p className={getElementStyleClassNames(["ContentPageCaptionTextSize",
                                              "CaptionText"])}
      >
        {props.caption}
      </p>
    </div>
  );
}
