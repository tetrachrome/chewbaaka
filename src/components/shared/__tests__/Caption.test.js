/**
 * Caption.test.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 09, 2020
 * Updated  : Jul 29, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import Caption from '../Caption'

const caption = "This is a caption";

test('renders Caption component', () => {
  render(
    <Caption
      caption={caption}
    />
  );

  const captionElement = screen.getByTestId("CaptionComponentTestId");
  expect(captionElement).toBeInTheDocument();
  expect(captionElement.textContent).toBe(caption);
});

test('Caption component snapshot', () => {
  const tree = renderer
    .create(
      <Caption
        caption={caption}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
