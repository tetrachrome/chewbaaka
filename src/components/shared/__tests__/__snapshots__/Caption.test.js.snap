// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption component snapshot 1`] = `
<div
  className="CaptionContainer CaptionContainer_DebugColor DebugWireframe"
  data-testid="CaptionComponentTestId"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="IconContainer IconContainer_DebugColor DebugWireframe"
  >
    <i
      aria-hidden="true"
      className="image icon"
      onClick={[Function]}
    />
  </div>
  <p
    className="ContentPageCaptionTextSize ContentPageCaptionTextSize_DebugColor DebugWireframe CaptionText CaptionText_DebugColor DebugWireframe"
  >
    This is a caption
  </p>
</div>
`;
